zhaquirks/xiaomi/aqara/motion_agl1.py

```diff
@@ -0,0 +1,192 @@
+    }
+    _VALID_ATTRIBUTES = {IasZone.AttributeDefs.zone_status.id}
+
+
+class OccupancySensingLocal(LocalDataCluster, OccupancySensing):
+    """Virtual cluster for OccupancySensing."""
+
+    _VALID_ATTRIBUTES = {OccupancySensing.AttributeDefs.occupancy.id}
+
+
+class OppleCluster(XiaomiAqaraE1Cluster):
+    """Aqara manufacturer cluster for the presence sensor FP1E."""
+
+    class AttributeDefs(BaseAttributeDefs):
+        """Manufacturer specific attributes."""
+
+        # The configurable maximum detection distance in millimeters (default 600 = 6 meters).
+        approach_distance = ZCLAttributeDef(
+            id=0x015B,
+            type=types.uint32_t,
+            access="rw",
+            is_manufacturer_specific=True,
+        )
+
+        # Detected motion
+        motion = ZCLAttributeDef(
+            id=0x0160,
+            type=AqaraMotion,
+            access="rp",
+            is_manufacturer_specific=True,
+        )
+
+        # Distance to the detected motion in millimeters
+        motion_distance = ZCLAttributeDef(
+            id=0x015F,
+            type=types.uint32_t,
+            access="rp",
+            is_manufacturer_specific=True,
+        )
+
+        # The configurable detection sensitivity
+        motion_sensitivity = ZCLAttributeDef(
+            id=0x010C,
+            type=AqaraMotionSensitivity,
+            access="rw",
+            is_manufacturer_specific=True,
+        )
+
+        # Detected occupancy
+        occupancy = ZCLAttributeDef(
+            id=0x0142,
+            type=AqaraOccupancy,
+            access="rp",
+            is_manufacturer_specific=True,
+        )
+
+        # Trigger AI spatial learning (write 1)
+        reset_no_presence_status = ZCLAttributeDef(
+            id=0x0157,
+            type=types.uint8_t,
+            access="w",
+            is_manufacturer_specific=True,
+        )
+
+        # Trigger device restart (write 0)
+        restart_device = ZCLAttributeDef(
+            id=0x00E8,
+            type=types.Bool,
+            access="w",
+            is_manufacturer_specific=True,
+        )
+
+    def _update_attribute(self, attrid: int, value: Any) -> None:
+        super()._update_attribute(attrid, value)
+        if attrid == self.AttributeDefs.occupancy.id:
+            self.endpoint.occupancy.update_attribute(
+                OccupancySensing.AttributeDefs.occupancy.id,
+                OccupancySensing.Occupancy.Occupied
+                if value == AqaraOccupancy.Occupied
+                else OccupancySensing.Occupancy.Unoccupied,
+            )
+        elif attrid == self.AttributeDefs.motion.id:
+            self.endpoint.ias_zone.update_attribute(
+                IasZone.AttributeDefs.zone_status.id,
+                IasZone.ZoneStatus.Alarm_1 if value == AqaraMotion.Moving else 0,
+            )
+
+
+(
+    QuirkBuilder("aqara", "lumi.sensor_occupy.agl1")
+    .friendly_name(model="Presence Sensor FP1E", manufacturer="Aqara")
```

Comment:
I know it's `model, manufacturer` in zigpy, but let's do it the other way around to match the order of `QuirkBuilder`.
Since they are keyword arguments here, the order technically doesn't matter. We should still consider swapping it around in zigpy to align with the typical `manufacturer, model` order though.
```suggestion
    .friendly_name(manufacturer="Aqara", model="Presence Sensor FP1E")
```